### PR TITLE
Fixes some bad reference stuff with player ships

### PR DESCRIPTION
**Pull request by Bokkiewokkie**
## About The Pull Request

Fixes some guns being added to the ship's weapon datums several thousands of times over and reference deletion for ship consoles.

## Why It's Good For The Game

Useless duplicate references are bad

## Changelog
:cl:
code: deleting ship consoles now dels their references properly.
code: stops ship weapons from filling the ship's loaded weapons list with their own refs.
/:cl:

### CODE REVIEW

File: nsv13/code/game/machinery/computer/helm.dm [hunk 1]
```diff
@@ -9,6 +9,10 @@
 /obj/machinery/computer/ship/helm/syndicate
 	req_one_access = list(ACCESS_SYNDICATE)
 
+/obj/machinery/computer/ship/helm/Destroy()
+	linked?.helm -= src
```
Comment (DeltaFire15) [inline nsv13/code/game/machinery/computer/helm.dm:13]: helm is just a object reference var, not a list var.
do something like if
(linked?.helm == src)
    linked.helm = null

File: nsv13/code/game/machinery/computer/tactical.dm [hunk 2]
```diff
@@ -6,6 +6,10 @@
 	position = "gunner"
 	circuit = /obj/item/circuitboard/computer/ship/tactical_computer
 
+/obj/machinery/computer/ship/tactical/Destroy()
+	linked?.tactical -= src
```
Comment (DeltaFire15) [inline nsv13/code/game/machinery/computer/tactical.dm:10]: Same thing here

File: nsv13/code/modules/overmap/weapons/ship_weapon.dm [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
 
+//These procs should *really* not be here
 /obj/structure/overmap/proc/add_weapon(obj/machinery/ship_weapon/weapon)
 	if(weapon_types[weapon.fire_mode])
 		var/datum/ship_weapon/SW = weapon_types[weapon.fire_mode]
 		SW.add_weapon(weapon) //hand it over to the datum for sane things like adding it idk
 
+/obj/structure/overmap/proc/remove_weapon(obj/machinery/ship_weapon/weapon)
+	var/datum/ship_weapon/SW = weapon_types[weapon.fire_mode]
+	SW.weapons -= weapon
```
Comment (DeltaFire15) [inline nsv13/code/modules/overmap/weapons/ship_weapon.dm:11]: weapons is a list with sublists so this won't do, remove it from the sublists instead
( SW.weapons["all"] -= weapon and SW.weapons["loaded"] -= weapon )

Also the proc is unused, you may want to add it to where you wanted it to be.

**Review by DeltaFire15 — CHANGES_REQUESTED** [review]

File: nsv13/code/game/machinery/computer/helm.dm (hunk 1 above)
Comment (Bokkiewokkie) [reply nsv13/code/game/machinery/computer/helm.dm:13]: Coder brain setting in... thanks for catching that

File: nsv13/code/modules/overmap/weapons/ship_weapon.dm (hunk 3 above)
Comment (Bokkiewokkie) [reply nsv13/code/modules/overmap/weapons/ship_weapon.dm:11]: Whoops, probably good to use it yeah!

File: nsv13/code/game/machinery/computer/helm.dm (hunk 1 above)
Comment (TheFakeElon) [reply nsv13/code/game/machinery/computer/helm.dm:13]: the fact that byond allows this... 😩

File: nsv13/code/modules/munitions/ship_weapons/_ship_weapon.dm [hunk 4]
```diff
@@ -372,7 +372,7 @@
 /obj/machinery/ship_weapon/proc/update()
 	if(weapon_type) // Who would've thought creating a weapon with no weapon_type would break everything!
 		if(!safety && chambered)
-			LAZYADD(weapon_type.weapons["loaded"] , src)
+			LAZYOR(weapon_type.weapons["loaded"] , src) //OR to avoid duplicating refs
```
Comment (TheFakeElon) [inline nsv13/code/modules/munitions/ship_weapons/_ship_weapon.dm:375]: weapons is not a lazy list afaik
```suggestion
			weapon_type.weapons["loaded"] |= src
```

File: nsv13/code/game/machinery/computer/helm.dm (hunk 1 above)
Comment (TheFakeElon) [reply nsv13/code/game/machinery/computer/helm.dm:13]: The fact that byond allows this.. 😫

**Comment by Bokkiewokkie** [comment]
Whaat there's no secret sprite changes in here github wdym

**Review by covertcorvid — APPROVED** [review]

**Pull request merged**